test/collectionUtils/InlineValueSetTypePredicate.test.ts

```diff
@@ -0,0 +1,81 @@
+import EntityDefinitionBuilder from "../model/dataTypes/EntityDefinitionBuilder";
+import EntityDefinition from "../../src/model/dataTypes/EntityDefinition";
+import InlineValueSetTypePredicate from "../../src/collectionUtils/InlineValueSetTypePredicate";
+import DataType from "../../src/model/dataTypes/DataType";
+import MemberVariable from "../../src/model/dataTypes/MemberVariable";
+
+describe("InlineValueSetTypePredicate", () => {
+  let predicate: InlineValueSetTypePredicate;
+  let entityDefinitionBuilder: EntityDefinitionBuilder;
+  let entity: EntityDefinition;
+  let elementType: DataType;
+  let otherType: DataType;
+  let wrongNs: DataType;
+  let systemStringType: DataType;
+  let systemDateType: DataType;
+  let valueMember: MemberVariable;
+  let wrongNameMember: MemberVariable;
+  let wrongTypeMember: MemberVariable;
+  let wrongSystemTypeMember: MemberVariable;
+
+  beforeEach(() => {
+    predicate = new InlineValueSetTypePredicate();
+
+    elementType = DataType.getInstance("FHIR", "Element", "/tmp");
+    otherType = DataType.getInstance("WrongNS", "Element", "/tmp");
+    wrongNs = DataType.getInstance("FHIR", "OtherType", "/tmp");
+    systemStringType = DataType.getInstance("System", "String", "/tmp");
+    systemDateType = DataType.getInstance("System", "Date", "/tmp");
+
+    entityDefinitionBuilder = new EntityDefinitionBuilder();
+    valueMember = new MemberVariable(systemStringType, "value");
+    wrongNameMember = new MemberVariable(systemStringType, "notValue");
+    wrongTypeMember = new MemberVariable(wrongNs, "value");
+    wrongSystemTypeMember = new MemberVariable(systemDateType, "value");
+    entityDefinitionBuilder.parentType = elementType;
+    entityDefinitionBuilder.memberVariables = [valueMember];
+  });
+
+  it("should return true if entity extends FHIR.Element and has single value member", () => {
+    entity = entityDefinitionBuilder.buildEntityDefinition();
+    expect(predicate.evaluate(entity)).toBeTrue();
+  });
+
+  it("should return false if entity has no parent type", () => {
+    entityDefinitionBuilder.parentType = null;
+    entity = entityDefinitionBuilder.buildEntityDefinition();
+    expect(predicate.evaluate(entity)).toBeFalse();
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5963561)